tests/functional/cylc-ping/04-check-keys-remote.t

```diff
@@ -16,48 +16,69 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #-------------------------------------------------------------------------------
 # Checks remote ZMQ keys are created and deleted on shutdown.
-export REQUIRE_PLATFORM='loc:remote comms:tcp'
+export REQUIRE_PLATFORM='loc:remote'
 . "$(dirname "$0")/test_header"
 set_test_number 5
 
-init_suite "${TEST_NAME_BASE}" <<'__FLOW_CONFIG__'
-#!jinja2
-[scheduler]
+create_test_global_config '' "
+[platforms]
+    [[$CYLC_TEST_PLATFORM]]
+        retrieve job logs = True
+"
+
+init_suite "${TEST_NAME_BASE}" <<__FLOW_CONFIG__
 [scheduling]
     [[graph]]
-        R1 = holder => held
+        R1 = keys
+
 [runtime]
-    [[holder]]
-        script = cylc hold "${CYLC_SUITE_NAME}"
-        platform = {{CYLC_TEST_PLATFORM}}
-    [[held]]
-        script = true
+    [[keys]]
+        platform = $CYLC_TEST_PLATFORM
+        script = """
+            find \
+                "\${CYLC_SUITE_RUN_DIR}" \
+                -type f \
+                -name "*key*" \
+                | awk -F/ '{print \$NF}'|sort > "\${CYLC_TASK_LOG_ROOT}-find-out"
+        """
+        [[[environment]]]
+            LANG = C
 __FLOW_CONFIG__
 
-run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}" \
-    -s "CYLC_TEST_PLATFORM='${CYLC_TEST_PLATFORM}'"
-suite_run_ok "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}" \
-    --debug -s "CYLC_TEST_PLATFORM='${CYLC_TEST_PLATFORM}'"
-RRUND="cylc-run/${SUITE_NAME}"
-RSRVD="${RRUND}/.service"
-poll_grep_suite_log 'Holding all waiting or queued tasks now'
-SSH='ssh -n -oBatchMode=yes -oConnectTimeout=5'
- 
-${SSH} "${CYLC_TEST_HOST}" \
-find "${RSRVD}" -type f -name "*key*"|awk -F/ '{print $NF}'|sort >'find.out'
+run_ok "${TEST_NAME_BASE}-validate" cylc validate \
+    "${SUITE_NAME}"
+suite_run_ok "${TEST_NAME_BASE}-run" cylc run \
+    "${SUITE_NAME}" \
+    --no-detach \
+    --debug
 
-sort >'keys'<<__OUT__
+KEYS_FILE="$(cylc cat-log -m p "$SUITE_NAME" 'keys.1' -f job-find-out)"
+if [[ "$CYLC_TEST_PLATFORM" == *shared* ]]; then
+    cmp_ok "$KEYS_FILE" <<__OUT__
+client.key_secret
 client_${CYLC_TEST_INSTALL_TARGET}.key
+server.key
+server.key_secret
+__OUT__
+else
+    cmp_ok "$KEYS_FILE" <<__OUT__
 client.key_secret
+client_${CYLC_TEST_INSTALL_TARGET}.key
 server.key
 __OUT__
```

Comment:
Different key files are present if the filesystem is shared, it's quite good to have a few dual-modal tests like this to ensure that the shared filesystem usage is working as intended.